LibRaw wavelet denoise options

ssh4net opened the pull request:
``raw:threshold``
     - float
     - Libraw parameter for noise reduction through wavelet denoising..
       The best threshold should be somewhere between 100 and 1000.
- ``raw:fbdd_noiserd`` - int - Controls FBDD noise reduction before demosaic. 0 - do not use FBDD noise reduction, 1 - light FBDD reduction, 2 (and more) - full FBDD reduction

<!-- This is just a guideline and set of reminders about what constitutes -->
<!-- a good PR. Feel free to delete all this matter and replace it with   -->
<!-- your own detailed message about the PR, assuming you hit all the     -->
<!-- important points made below.                                         -->


## Description

<!-- Please provide a description of what this PR is meant to fix, and  -->
<!-- how it works (if it's not going to be very clear from the code).   -->

## Tests

<!-- Did you / should you add a testsuite case (new test, or add to an  -->
<!-- existing test) to verify that this works?                          -->


## Checklist:

<!-- Put an 'x' in the boxes as you complete the checklist items -->

- [ ] I have read the [contribution guidelines](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/master/CONTRIBUTING.md).
- [ ] I have updated the documentation, if applicable.
- [ ] I have ensured that the change is tested somewhere in the testsuite
  (adding new test cases if necessary).
- [ ] If I added or modified a C++ API call, I have also amended the
  corresponding Python bindings (and if altering ImageBufAlgo functions, also
  exposed the new functionality as oiiotool options).
- [ ] My code follows the prevailing code style of this project. If I haven't
  already run clang-format before submitting, I definitely will look at the CI
  test that runs clang-format and fix anything that it highlights as being
  nonconforming.

ssh4net commented:
@lgritz nothing special just two options:
raw:threshold and raw:fbdd_noiserd
If the first one clearly visible results. Recommended to use values from 100.00 to 1000.00 

but fbdd_noiserd I did not get how it works in libraw so just added it as it is also about denoising (1 or 2 should enable different modes of this pre-demosaic denoiser).

My main concern, that compared to direct libraw demosaic that not affect processing time a lot. OpenImageIO has 2-3 times slowdown on wavelet demosaic,
But i don't see anything suspicious except the debugger 3 or 4 times passing this libraw options. First time on metadata read, and set nothing but last three (or more?) times it already has actual settings set.
But worth to mention, looks like demosaic did only once. So i can't get where this slowdown in oiio happen.

lgritz commented on src/doc/builtinplugins.rst:
```diff
@@ -1926,8 +1926,15 @@ options are supported:
-
+   * - ``raw:threshold``
+     - float
+     - Libraw parameter for noise reduction through wavelet denoising.. 
```
Minor correction: `..` -> `.`

lgritz commented:
Check the CI run to see that clang-format found a couple of very minor things, but other than that, it looks good to me.

ssh4net commented:
Fixed.
I also added info about default values.

Do you have any ideas why enabling this pre-demosaic denoising can have such a significant impact on plugin speed when libraw directly has only 5-10% slowdown?

lgritz commented on src/doc/builtinplugins.rst:
```diff
@@ -1926,8 +1926,17 @@ options are supported:
+     - Controls FBDD noise reduction before demosaic.
+       0 - do not use FBDD noise reduction, 1 - light FBDD reduction,
+       2 (and more) - full FBDD reduction
+       (Default: 0.0)
```
I'm sorry, last thing, I promise! Since this one is an integer parameter, let's document the default as 0 instead of 0.0.

lgritz replied:
```suggestion
       (Default: 0)
```

lgritz reviewed (approved):
LGTM!

(I fixed the 0.0 -> 0 suggestion for you)
